rs-react-app/src/components/Card.tsx

```diff
@@ -1,19 +1,62 @@
+import { useSelectedItemsStore } from '../store/selectedItemsStore';
+
 interface CardProps {
-  name?: string;
-  description?: string;
+  id: number;
+  name: string;
+  status: string;
+  species: string;
+  gender: string;
   image?: string;
 }
 
-export const Card = ({ name, description, image }: CardProps) => {
+export const Card = ({
+  id,
+  name,
+  status,
+  species,
+  gender,
+  image,
+}: CardProps) => {
+  const { selectedItems, toggleItem } = useSelectedItemsStore();
+  const isSelected = selectedItems.some((i) => i.id === id);
```

Comment:
Subscribing to the whole selectedItems array forces all cards to re-render on any change.
It's better to subscribe only to a boolean and the single action
```suggestion
const isSelected = useSelectedItemsStore((s) => s.selectedItems.some((i) => i.id === id));
const toggleItem = useSelectedItemsStore((s) => s.toggleItem);
```
Otherwise you are subscribing to the whole store and component will be rerendered on every change

You can check [this guide](https://zustand.docs.pmnd.rs/guides/auto-generating-selectors) to automatically create selectors